strawberryfields/gbs/sample.py

```diff
@@ -332,37 +370,71 @@ def gaussian(
     if not 0 <= loss <= 1:
         raise ValueError("Loss parameter must take a value between zero and one")
 
-    n_modes = len(alpha)
+    n_modes = len(t)
+
+    # initialize engine
 
     eng = sf.LocalEngine(backend="gaussian")
-    gbs = sf.Program(n_modes)
 
+    # initialize program objects
+
+    if np.all(t != 0):
+        gbs = sf.Program(n_modes * 2)
+    else:
+        gbs = sf.Program(n_modes)
+
+    # start circuit
     # pylint: disable=expression-not-assigned,pointless-statement
     with gbs.context as q:
 
-        # this interferometer has no action, but is kept to follow the decomposition given in the
-        # docstring
-        sf.ops.Interferometer(U1) | q
+        # two-mode squeezing gates
+
+        if np.all(t != 0):
```

Comment:
The code below this `if` statement can probably go under the `if` of line 381.  You then probably need to also add `with gbs.context as q:` under both the `if` on 381 and the `else` on 383